src/components/IconState.vue

```diff
@@ -50,23 +47,39 @@ export default {
       localIcon: this.icon,
       dialog: false,
       previousIcon: this.icon,
+      textFieldRules: [
+        (v) => !!v || 'icon name is required!',
+        (v) => this.$_validateIconName(v) || 'Not a valid icon name',
+      ],
     };
   },
   methods: {
+    $_validateIconName() {
+      const icon1 = document.querySelector('.local-icon [name="selected-icon"]');
```

Comment:
The variable names in this method should be semantic. What does `icon1` mean? What does `objContent` mean? What object and what kind of content is in `objContent`?